### DISCUSSION TITLE
importing db from another netbox container instance

### DISCUSSION THREAD

**Discussion by sbj-ee** [netbox-community/netbox#6192] [Q&A] [answered]
I have been attempting to import my netbox db that currently exists on one container into a different netbox container on a new server.

docker exec -t fe65ab1299aa pg_dumpall -c -U netbox > ./dump_`date +%d-%m-%Y"_"%H_%M_%S`.sql

I successfully dump the netbox db

I have not had any success importing it though. Any tips or clues?

root@Netbox:~/netbox-docker# psql netbox < ./dump_15-04-2021_23_57_23.sql 
psql: error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

root@Netbox:~/netbox-docker# docker exec -t fe65ab1299aa psql netbox < ./dump_15-04-2021_23_57_23.sql 
psql: error: FATAL:  role "root" does not exist

root@Netbox:~/netbox-docker# docker exec -t fe65ab1299aa psql -U netbox netbox < ./dump_15-04-2021_23_57_23.sql 
psql (12.6)
Type "help" for help.

netbox=#                    (I cannot do anything with this connection)

**Comment by devon-mar** [comment, marked as answer]
Take a look at the [troubleshooting page](https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting) from the netbox-docker repo.

**Reply by sbj-ee** [reply]
I figured it out, but I'm fighting postgres...
I got some of the data into the database, but I really need to drop the netbox db so that the insert is fully successful.
I'm just trying to figure out how to drop the database as it keeps complaining 

## see the complaining
root@Netbox:~# docker exec -i netbox-docker_postgres_1 psql --username netbox netbox < ./dump_15-04-2021_23_57_23.sql
ERROR:  cannot drop the currently open database
SET
SET
SET
ERROR:  current user cannot be dropped
ERROR:  role "netbox" already exists
ALTER ROLE
SET
SET
SET
. . . 


root@Netbox:~# docker exec -i netbox-docker_postgres_1 psql --username netbox -c "drop database netbox;"
ERROR:  cannot drop the currently open database

I'll check out the troubleshooting page too

**Reply by devon-mar** [reply]
Are you able to just recreate the postgres container?

**Reply by sbj-ee** [reply]
I figured it out.. I had to connect to the postgres db and then drop the netbox db.
Then I could successfully insert all of my data from my netbox pg_dump from my other machine.

I'm rocking now... woohoo

**Comment by sbj-ee** [comment]
```
root@Netbox:~/netbox-docker# docker-compose stop netbox netbox-worker
root@Netbox:~/netbox-docker# docker-compose exec postgres sh -c 'psql -U $POSTGRES_USER postgres'

root@Netbox:~/netbox-docker# docker-compose exec postgres sh -c 'psql -U $POSTGRES_USER postgres'
psql (12.6)
Type "help" for help.

postgres=# drop database netbox;

postgres=# \l
                              List of databases
   Name    | Owner  | Encoding |  Collate   |   Ctype    | Access privileges 
-----------+--------+----------+------------+------------+-------------------
 postgres  | netbox | UTF8     | en_US.utf8 | en_US.utf8 | 
 template0 | netbox | UTF8     | en_US.utf8 | en_US.utf8 | =c/netbox        +
           |        |          |            |            | netbox=CTc/netbox
 template1 | netbox | UTF8     | en_US.utf8 | en_US.utf8 | netbox=CTc/netbox+
           |        |          |            |            | =c/netbox
(3 rows)

postgres=# create database netbox;

root@Netbox:~/netbox-docker# gunzip -c ../dump_15-04-2021_23_57_23.sql.gz | docker-compose exec -T postgres sh -c 'psql -U $POSTGRES_USER $POSTGRES_DB'

...and your data should be there
```

**Reply by sbj-ee** [reply]
One more thing ...  Now I need to get all of my images (media) onto my new machine so they show nicely in the racks on the new netbox instance

**Reply by sbj-ee** [reply]
uggg  - ok... I don't have too many images and I still have them on my Macbook. So I'm just going to re-associate the images by uploading them.

```
bash-5.1$ du -a .
4	./image-attachments/.gitignore
8	./image-attachments
24	./devicetype-images/HP_DL380_G10_-_front.png
16	./devicetype-images/Dell_R620_-_front.png
32	./devicetype-images/sencore-txs3600-front.png
52	./devicetype-images/Apex1000-rear.png
56	./devicetype-images/Apex1000-front.png
68	./devicetype-images/Ciena_z77_-_front.png
36	./devicetype-images/arris-dsr6100-front.png
60	./devicetype-images/monroe-rear.png
48	./devicetype-images/nexus-9000-93180.png
32	./devicetype-images/cisco-d9858-front.png
24	./devicetype-images/Arris_xms_Mini_-_front.png
76	./devicetype-images/OM2000-front.png
28	./devicetype-images/cisco-2960-front.png
40	./devicetype-images/cisco-d9828.png
4	./devicetype-images/.gitignore
36	./devicetype-images/cisco-d9858-rear.png
28	./devicetype-images/BNP_3xr_-_front.png
36	./devicetype-images/arris-dsr4440-rear.png
40	./devicetype-images/arris-dsr7401-rear.png
52	./devicetype-images/monroe-front.png
28	./devicetype-images/quintech-front.png
12	./devicetype-images/cisco-3650-front.png
16	./devicetype-images/Dell_R620_-_rear.png
540	./devicetype-images/sonifex-ps-sends-rear.png
260	./devicetype-images/prostream1000-rear.png
124	./devicetype-images/NCS5508-front.png
64	./devicetype-images/arris-4410md.png
44	./devicetype-images/arris-dsr6100-rear.png
32	./devicetype-images/arris-4410-rear.png
96	./devicetype-images/AGB240-front.png
12	./devicetype-images/BNP_2xr_-_front.png
28	./devicetype-images/viavi-front.png
32	./devicetype-images/arris-dsr7401-front.png
68	./devicetype-images/sandvine-pts-32000-front.png
384	./devicetype-images/prostream1000-front.png
16	./devicetype-images/HP_DL380p_Gen8_-_front.png
120	./devicetype-images/AGB240-rear.png
116	./devicetype-images/ARPD1000-front_6PKXKqj.png
40	./devicetype-images/HP_Apollo_4200_Gen10.png
468	./devicetype-images/sonifex-ps-sends-front.png
8	./devicetype-images/Cisco_PowerVu_9800_-_front.png
24	./devicetype-images/Arris_E6000_-_front.png
32	./devicetype-images/arris-dsr4440-front.png
24	./devicetype-images/arris-4410.png
40	./devicetype-images/Cisco_9504_-_front.png
84	./devicetype-images/gi-rpd2000.png
36	./devicetype-images/cisco-d9854.png
52	./devicetype-images/sencore-txs3600-rear.png
40	./devicetype-images/quintech-rear.png
48	./devicetype-images/Cisco_871_-_front.png
52	./devicetype-images/Miranda_server_-_front.png
84	./devicetype-images/Evertz-rear.png
16	./devicetype-images/cisco-3750_-_front.png
144	./devicetype-images/ARPD1000-rear.png
60	./devicetype-images/HP_Apollo_4200_Gen10_rear.png
44	./devicetype-images/Aurora_CH3000_-_front.png
24	./devicetype-images/Calix_E7_-_front.png
4104	./devicetype-images
4116	.
bash-5.1$ pwd
/opt/netbox/netbox/media
bash-5.1$ cd ../..
bash-5.1$ tar -czf netbox_media.tar.gz netbox/media/
tar: can't open 'netbox_media.tar.gz': Permission denied
```

**Reply by devon-mar** [reply]
Maybe the directory your trying to make the tar isn't readable? 

For your new deployment, you might want to mount `media` or use a docker volume.

**Reply by sbj-ee** [reply]
It's readable, but owned by root - which I'm not going into the bash shell of the netbox container.
As long as I got all of my devices, IPs, and DCIM info... I'm happy.

Yes - my new server has the storage on a RAID array - even easier I just take snapshots of my container using proxmox.
I was initially just trying out netbox on my Macbook, then when I started liking it and actually had a decent amount of data - it was time to move it onto a server.
